app/pages/privacy-policy.spec.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import assert from 'assert';
+import { shallow } from 'enzyme';
+import PrivacyPolicy from './privacy-policy';
+import englishTranslations from '../locales/en';
+
+describe('PrivacyPolicy', () => {
+  let wrapper;
+
+  before(() => {
+    wrapper = shallow(<PrivacyPolicy />);
+  });
+
+  it('renders without crashing', () => {
+    const privacyPolicyContainer = wrapper.find('div.content-container');
+    assert.equal(privacyPolicyContainer.length, 1);
+  });
+
+  describe('heading', () => {
+    it('renders a <Translate /> component', () => {
+      assert.equal(wrapper.find('Translate').length, 1);
+    });
+
+    it('renders page title content', () => {
+      assert.equal(wrapper.find('Translate').prop('content'), 'privacy.title');
+    });
+  });
+
+  describe('user agreement section', () => {
+    let markdownElements;
+
+    before(() => {
+      markdownElements = wrapper.find('div.column').first().children();
+    });
+
+    it('renders four markdown elements', () => {
+      assert.equal(markdownElements.length, 4);
+    });
+
+    it('renders each of its sections', () => {
+      const userAgreementSections = Object.keys(englishTranslations.privacy.userAgreement);
+
+      markdownElements.forEach((element, index) => {
+        assert.ok(element.props().children.match(userAgreementSections[index]));
```

Comment:
On my first pass I didn't quite understand what you were trying to test, but it's more clear now. I think you're trying to test that the english translation gets rendered in the DOM? However, that isn't what is happening. What you're asserting here is that the the child of the Markdown component is matching a regex. You're giving it the key from the locales file, so it'll try to match literally for instance `'en.privacy.userAgreement.summary'`. It's passing because the shallow render output is actually throwing a counterpart error, for example: `missing translation: en.privacy.userAgreement.summary`. I'm not totally sure but it might be because shallow rendering doesn't run the full react component lifecycle.

Reply:
Yeah, I did see the counterpart error, and my thought process was that matching on the regex would test whether each of the markdown elements at least text-matched a key in the correct part of the translations. so if something changes and the markdown element doesn't match what's in the translation key, the test will fail. Are you ok with this approach? or is it outside the responsibility of the tests for this component?

Reply:
It's not obvious at a cursory glance that the reason it passes is because counterpart is failing in the shallow render. I think this could be confusing to other devs who may look at this later. Ultimately, I don't think we don't want to test if something is failing accidentally. You'd need to fix the test rendering so that what we actually expect, the parsed markdown into HTML with the expected English text, and then make an assertion that the rendered HTML and text equals the expected HTML and text. Or refactor the tests to actually test just the component as I suggest, not its children or the dependencies.

Testing the specific rendered text after all the parsing and transpiling done by Markdownz, markdown-it, counterpart, and React is not something we should be doing since that is essentially retesting dependencies that are already tested.